File: front-end/src/services/api/events.ts
/**
 * Funções para chamadas de API relacionadas a eventos e métricas
 * Usa o cliente HTTP centralizado
 */

import { api } from './client';

// Tipos TypeScript (alinhados com o backend)
export interface Event {
  id: string;
  event_name: string;
  event_type: string;
  event_date: string;
  event_location: string;
  total_photos: number;
  total_athletes_estimated: number;
  status: string;
  created_at: string;
  processed_at?: string;
  metadata?: Record<string, unknown>;
}

export interface EventListResponse {
  events: Event[];
  total: number;
  limit: number;
  offset: number;
  has_more: boolean;
}

export interface BrandSummary {
  event_id: string;
  event_name: string;
  event_date: string;
  brand: string;
  persons_with_brand: number;
  total_items: number;
  brand_share_percent: number;
  person_coverage_percent: number;
}

export interface ProductSummary {
  event_id: string;
  event_name: string;
  product_type: string;
  persons_with_product: number;
  total_items: number;
  product_share_percent: number;
}

export interface DashboardMetrics {
  total_events: number;
  total_photos_analyzed: number;
  total_athletes_identified: number;
  total_brands_tracked: number;
}

/**
 * Lista todos os eventos com paginação
 */
export async function getEvents(
  limit: number = 100,
  offset: number = 0
): Promise<EventListResponse> {
  return api.get<EventListResponse>('/api/events', { limit, offset });
}

/**
 * Busca detalhes de um evento específico
 */
export async function getEventById(eventId: string): Promise<Event> {
  return api.get<Event>(`/api/events/${eventId}`);
}

/**
 * Busca resumo de marcas de um evento
 */
export async function getEventBrands(
  eventId: string
): Promise<{ event_id: string; brands: BrandSummary[] }> {
  return api.get<{ event_id: string; brands: BrandSummary[] }>(
    `/api/events/${eventId}/brands`
  );
}

/**
 * Busca resumo de produtos de um evento
 */
export async function getEventProducts(
  eventId: string
): Promise<{ event_id: string; products: ProductSummary[] }> {
  return api.get<{ event_id: string; products: ProductSummary[] }>(
    `/api/events/${eventId}/products`
  );
}

/**
 * Busca KPIs agregados para o dashboard
 */
export async function getDashboardMetrics(): Promise<DashboardMetrics> {
  return api.get<DashboardMetrics>('/api/metrics/dashboard');
}

File: app/core/events.py
"""
Lógica de negócio para eventos e métricas
Camada que orquestra serviços e aplica regras de negócio
"""

from typing import List, Dict, Any, Optional
from app.services.database import DatabaseService


class EventsService:
    """Serviço de lógica de negócio para eventos"""
    
    def __init__(self, db_service: DatabaseService):
        self.db = db_service
    
    def list_events(self, limit: int = 100, offset: int = 0) -> Dict[str, Any]:
        """
        Lista eventos com paginação
        Retorna eventos ordenados por data (mais recentes primeiro)
        """
        events = self.db.get_events(limit=limit, offset=offset)
        total = self.db.count_events()
        
        return {
            "events": events,
            "total": total,
            "limit": limit,
            "offset": offset,
            "has_more": (offset + limit) < total
        }
    
    def get_event_details(self, event_id: str) -> Optional[Dict[str, Any]]:
        """
        Busca detalhes completos de um evento
        Inclui informações básicas do evento
        """
        event = self.db.get_event_by_id(event_id)
        return event
    
    def get_event_brands(self, event_id: str) -> List[Dict[str, Any]]:
        """
        Retorna resumo de marcas de um evento
        Ordenado por brand_share_percent (maior para menor)
        """
        brands = self.db.get_brand_summary(event_id)
        return brands
    
    def get_event_products(self, event_id: str) -> List[Dict[str, Any]]:
        """
        Retorna resumo de produtos de um evento
        Ordenado por product_share_percent (maior para menor)
        """
        products = self.db.get_product_summary(event_id)
        return products
    
    def get_dashboard_metrics(self) -> Dict[str, Any]:
        """
        Retorna KPIs agregados para o dashboard
        Calcula métricas gerais do sistema
        """
        metrics = self.db.get_dashboard_metrics()
        
        # Adicionar métricas calculadas
        # Precisão média (placeholder - pode ser calculada depois)
        metrics["avg_accuracy"] = 94.7  # TODO: Calcular baseado em dados reais
        
        # Tempo médio de processamento (placeholder)
        metrics["avg_processing_time"] = 2.3  # TODO: Calcular baseado em dados reais
        
        return metrics

